converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs

```diff
@@ -100,5 +102,27 @@ private void AddAccessionNumber(ImagingStudy imagingStudy, DicomDataset dataset)
                 }
             }
         }
+
+        private void AddNumberOfSeries(ImagingStudy imagingStudy, DicomDataset dataset)
+        {
+            if (dataset.TryGetSingleValue(DicomTag.NumberOfStudyRelatedSeries, out int numberOfSeries))
```

Comment:
for now the worker is running single threaded, how do we handle out of order instance update?